pkg/odo/cli/dev/dev.go

```diff
@@ -193,29 +237,64 @@ func (o *DevHandler) RegenerateAdapterAndPush(pushParams common.PushParameters,
 }
 
 func regenerateComponentAdapterFromWatchParams(parameters watch.WatchParameters) (common.ComponentAdapter, error) {
-
-	// Parse devfile and validate. Path is hard coded because odo expects devfile.yaml to be present in the pwd/cwd.
-
 	devObj, err := ododevfile.ParseAndValidateFromFile(location.DevfileLocation(""))
 	if err != nil {
 		return nil, err
 	}
 
+	changed, err := haveEndpointsChanged(parameters.InitialDevfileObj, devObj)
+	if err != nil {
+		return nil, err
+	} else if changed {
+		fmt.Printf("\ntotal number of endpoints in the devfile have changed; please run `odo dev` again\n\n")
+	}
+
 	platformContext := kubernetes.KubernetesContext{
 		Namespace: parameters.EnvSpecificInfo.GetNamespace(),
 	}
 
 	return adapters.NewComponentAdapter(parameters.ComponentName, parameters.Path, parameters.ApplicationName, devObj, platformContext)
+}
+
+func haveEndpointsChanged(oldDevfile, newDevfile parser.DevfileObj) (bool, error) {
```

Comment:
@kadel at the moment, we are checking only if endpoints have been added/removed. Do we instead want to notify the user to do `odo dev` again when anything in devfile changes? Would change in env vars also require user to redo `odo dev`?